### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by chezou**
- Split API reference doc into module groups
- Change version number style to set implicit anchor in rest format

### CODE REVIEW

File: docs/api/api.rst [hunk 1]
```diff
@@ -0,0 +1,108 @@
+API
+=====
+
+:class:`tdclient.api.API` class is a internal class represents API.
```
Comment (takuti) [inline docs/api/api.rst]: typo: s/a internal/an internal/

**Comment by takuti** [comment]
One more thing - don't you set `currentmodule`? Feels like so many `tdclient` is just annoying...

```
.. currentmodule:: tdclient

models
----------------------

.. autodata:: models.BulkImport
.. autodata:: models.Database
.. autodata:: models.Schema
.. autodata:: models.Job
.. autodata:: models.Result
.. autodata:: models.ScheduledJob
.. autodata:: models.Schedule
.. autodata:: models.Table
.. autodata:: models.User
```

Screen Shot 2019-09-27 at 11 39 24 AM

**Comment by chezou** [comment]
@takuti Seems that `currentmodule` directive is a feature from autosummary, currently, we don't use it yet.

File: docs/api/api.rst (hunk 1 above)
Comment (chezou) [reply docs/api/api.rst]: thanks! da85488

**Pull request merged**